routes/api.php

```diff
@@ -14,6 +15,30 @@
 |
 */
 
+// Get all todo routes
+Route::get('todos', [TodoController::class, 'index']);
+
+// Create new todo
+Route::post('todos/new/{userId}', [TodoController::class, 'create']);
+
+// Get todo by id
+Route::get('todos/get/{id}', [TodoController::class, 'show']);
+
+// Delete a todo by id
+Route::delete('todos/delete/{id}', [TodoController::class, 'destroy']);
+
+// Update isChecked property of todo by id
+Route::put('todos/update/isChecked/{id}', [TodoController::class, 'update']);
+
+// Update title, description, and priority
+Route::put('todos/update/{id}', [TodoController::class, 'update']);
+
+// Search functionality for todos
+Route::post('todos/search/{userId}', [TodoController::class, 'search']);
```

Comment:
For the searching and sorting please use the index method of the TodoController.
And use the `/api/todos` API endpoint for that.